crates/solidity/outputs/npm/crate/Cargo.toml

```diff
@@ -1,18 +1,29 @@
 [package]
-name = "solidity_npm_crate"
+name = "slang_solidity_node_addon"
 version.workspace = true
 rust-version.workspace = true
 edition.workspace = true
 publish = false
 
+description = "slang_solidity compiled as a cdylib crate, to be used as Node.js addon."
+
+# At the time of writing, building cdylib/plugins with Cargo has limited support
+# (see <https://github.com/rust-lang/cargo/issues/8628> for the general tracking issue).
+# To workaround some of the issues, we create a synthetic crate that shares the same code as the original Rust crate
+# but only exposes a `cdylib` library.
+# There are a couple of things at play here:
+# - To build a Node addon, we need to build a standalone cdylib, that's set up to expect host `napi_*` functions at runtime
```

Comment:
I filed this bug to support pasing `--lib` to `napi build` to fix this: https://github.com/napi-rs/napi-rs/issues/1906

Thanks for investigating!